src/logging.c

```diff
@@ -185,44 +184,41 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
       case RCUTILS_GET_ENV_ONE:
         g_output_stream = stdout;
         break;
+      default:
+        fprintf(
+          stderr, "Invalid return from environment fetch\n");
+        RCUTILS_SET_ERROR_MSG(
+          "Invalid return from environment fetch");
+        return RCUTILS_RET_ERROR;
```

Comment:
I'm not a huge fan of setting the error state and printing to stderr. In my opinion, libraries shouldn't print to stderr unless there's no other way to raise an issue. This function can raise an issue with the return code and error state.

Again, I know this is the case already in this file, but I felt I should bring it up as long as we're looking at this file.